modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/plugins/fields-sidebar/components/FieldsSidebar.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayForm from '@clayui/form';
+import classNames from 'classnames';
+import {
+	EVENT_TYPES as CORE_EVENT_TYPES,
+	useConfig,
+	useForm,
+	useFormState,
+} from 'dynamic-data-mapping-form-renderer';
+import React, {useState} from 'react';
+
+import {
+	EDIT_CUSTOM_OBJECT_FIELD,
+	UPDATE_DATA_DEFINITION_AVAILABLE_LANGUAGE,
+	dropLayoutBuilderField,
+} from '../../../../js/actions.es';
+import Sidebar from '../../../../js/components/sidebar/Sidebar.es';
+import {EVENT_TYPES} from '../../../eventTypes';
+import FieldsSidebarBody from './FieldsSidebarBody';
+import FieldsSidebarSettingsBody from './FieldsSidebarSettingsBody';
+import FieldsSidebarSettingsHeader from './FieldsSidebarSettingsHeader';
+
+const sortFieldTypes = (fieldTypes) =>
+	fieldTypes.sort(({displayOrder: a}, {displayOrder: b}) => a - b);
+
+export const DataEngineFieldsSidebar = ({title}) => {
+	const {
+		activePage,
+		config,
+		customFields,
+		dataDefinition,
+		dataLayout,
+		defaultLanguageId,
+		editingLanguageId,
+		focusedCustomObjectField,
+		focusedField,
+		pages,
+	} = useFormState();
+	const {fieldTypes} = useConfig();
+	const dispatch = useForm();
+
+	const hasFocusedCustomObjectField = (focusedCustomObjectField) => {
+		return !!focusedCustomObjectField.settingsContext;
+	};
+
+	const hasFocusedField = Object.keys(focusedField).length > 0;
+
+	const displaySettings =
+		hasFocusedCustomObjectField(focusedCustomObjectField) ||
+		hasFocusedField;
+
+	const sortedFieldTypes = sortFieldTypes(
+		fieldTypes.filter(({group}) => group === 'basic')
+	);
+
+	return (
+		<FieldsSidebar
+			config={config}
+			customFields={customFields}
+			dataLayout={dataLayout}
+			defaultLanguageId={defaultLanguageId}
+			dispatchEvent={(type, payload) => {
+				if (type === 'field_change') {
+					const {editingLanguageId} = payload;
+
+					if (
+						!dataDefinition.availableLanguageIds.includes(
+							editingLanguageId
+						)
+					) {
+						dispatch({
+							payload: editingLanguageId,
+							type: UPDATE_DATA_DEFINITION_AVAILABLE_LANGUAGE,
+						});
+					}
+
+					if (hasFocusedCustomObjectField(focusedCustomObjectField)) {
+						dispatch({payload, type: EDIT_CUSTOM_OBJECT_FIELD});
+
+						return;
+					}
+				}
```

Comment:
We can ignore this for now, because some of these behaviors no longer make sense within the new base, because some of these do not need to be brought to this new side or because this event will be triggered by another location with another name.